form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -100,4 +102,31 @@ class FormFlowService(
     fun deleteByKey(definitionKey: String) {
         formFlowDefinitionRepository.deleteAllByIdKey(definitionKey)
     }
+
+    fun getBreadcrumbs(instance: FormFlowInstance): List<FormFlowBreadcrumb> {
+        val lastCompletedOrder = instance.getHistory()
+            .filter { it.submissionData != null }
+            .maxByOrNull { it.submissionOrder }
+            ?.order ?: -1
+        val historicBreadcrumbs = instance.getHistory()
+            .map { FormFlowBreadcrumb.of(it, it.order <= lastCompletedOrder + 1, it.order <= lastCompletedOrder) }
+        val futureBreadcrumbs = getFutureSteps(instance)
+            .map { FormFlowBreadcrumb.of(it) }
+        return historicBreadcrumbs + futureBreadcrumbs
+    }
+
+    private fun getFutureSteps(instance: FormFlowInstance): List<FormFlowStep> {
+        return getFutureSteps(instance.getHistory().last().definition)
+    }
+
+    private fun getFutureSteps(step: FormFlowStep, result: MutableList<FormFlowStep> = mutableListOf()): List<FormFlowStep> {
+        step.nextSteps.forEach { nextStep ->
+            val featureStep = step.id.formFlowDefinition!!.steps.first { it.id.key == nextStep.step }
```

Comment:
So the first step will be the happy path? for every form flow step? I'm not sure if that is explicit enough.

Reply:
Also, shouldn't this be `futureStep` rather than `featureStep`?

Reply:
The first step is indeed the happy path. This is explained in the documentation.

Also: fixed typo